packages/go/cypher/model/pgsql/translate/expression.go

```diff
@@ -0,0 +1,230 @@
+package translate
+
+import (
+	"fmt"
+	"github.com/specterops/bloodhound/cypher/model/pgsql"
+)
+
+func operandsToBinaryExpression(operator pgsql.Operator, rightOperand, leftOperand pgsql.Expression) *pgsql.BinaryExpression {
+	binaryExpression := &pgsql.BinaryExpression{
+		Operator: operator,
+	}
+
+	// Inspect left operand
+	switch typedOperand := leftOperand.(type) {
+	case *pgsql.BinaryExpression:
+		binaryExpression.LOperand = typedOperand
+		binaryExpression.LDependencies = typedOperand.CombinedDependencies()
+
+	case pgsql.Identifier:
+		binaryExpression.LOperand = typedOperand
+		binaryExpression.LDependencies = map[pgsql.Identifier]struct{}{
+			typedOperand: {},
+		}
+
+	case pgsql.CompoundIdentifier:
+		binaryExpression.LOperand = typedOperand
+		binaryExpression.LDependencies = map[pgsql.Identifier]struct{}{
+			typedOperand.Root(): {},
+		}
+
+	default:
+		binaryExpression.LOperand = typedOperand
+		binaryExpression.LDependencies = map[pgsql.Identifier]struct{}{}
+	}
+
+	// Inspect right operand
+	switch operand := rightOperand.(type) {
+	case *pgsql.BinaryExpression:
+		binaryExpression.ROperand = operand
+		binaryExpression.RDependencies = operand.CombinedDependencies()
+
+	case pgsql.Identifier:
+		binaryExpression.ROperand = operand
+		binaryExpression.RDependencies = operandIdentifiers(operand)
+
+	case pgsql.CompoundIdentifier:
+		binaryExpression.ROperand = operand
+		binaryExpression.RDependencies = map[pgsql.Identifier]struct{}{
+			operand.Root(): {},
+		}
+
+	default:
+		binaryExpression.ROperand = operand
+		binaryExpression.RDependencies = map[pgsql.Identifier]struct{}{}
+	}
+
+	return binaryExpression
+}
+
+func operandIdentifiers(operand pgsql.Expression) map[pgsql.Identifier]struct{} {
+	switch typedOperand := operand.(type) {
+	case *pgsql.BinaryExpression:
+		return typedOperand.CombinedDependencies()
+
+	case pgsql.Identifier:
+		return map[pgsql.Identifier]struct{}{
+			typedOperand: {},
+		}
+
+	case pgsql.CompoundIdentifier:
+		return map[pgsql.Identifier]struct{}{
+			typedOperand.Root(): {},
+		}
+
+	default:
+		return map[pgsql.Identifier]struct{}{}
+	}
+}
+
+type ExpressionTreeTranslator struct {
```

Comment:
could you provide a minimal example of how to use this structure?  perhaps as a test case in a new expression_test.go file?  i can extend off of your initial example with more coverage, but im curious how this machinery is intended to be used.  

seeing it work in isolation would be illuminating, but also not sure if its possible to reason about in isolation. what do you think?